net/portmapper/portmapper.go

```diff
@@ -279,6 +281,9 @@ func (c *Client) Close() error {
 	}
 	c.closed = true
 	c.invalidateMappingsLocked(true)
+	if c.updates != nil {
+		c.updates.Close()
```

Comment:
hmm, should we hold onto the *Client and close that too? As it stands, the Client will hang around until the entire bus is closed, with 0 publishers/subscribers. In practice doesn't matter hugely in tailscaled, but it'd be nice to tear down everything that was made at New() time.

Reply:
I thought about it, and decided that probably is OK, since the lifecycle should be the same for all of them. But if you think it's worthwhile it's cheap to add another field.

Reply:
This makes me wonder if we shouldn't do a `runtime.AddCleanup` on clients when they are constructed, since only the holder of the client should have had the ability to add/remove publishers and subscribers anyway.

Reply:
(I added the field for now, we can change it later if we want to)

Reply:
I think my main worry is tests that check for goroutine leaks, since once a Client gets any subscriber it spins up a routine for dispatch. So I think I'd like to ensure explicit orderly cleanup for that reason.

I hadn't considered runtime.AddCleanup. I _think_ the reasoning for goroutine leaks applies there too, in that it'd be convenient to have stuff tidy itself up "later", but leaves a bear trap for later when we add a subscriber to a client and tests start blowing up.

So, I think my default would be to err on the side of explicitly closing Clients for now, although this maybe points to a deficiency in the API design we should look at (e.g. refcount clients somehow, or something?...). But that should probably be a separate change.

Reply:
I think if we get to the point refcounting, we'd be better off doing some kind of hybrid "pubclient / subclient" convenience wrapper, that someone can create to manage the two as a pair for the common(?) case of "I want to publish/subscribe a single event type in and not have to think about separate values".

Anecdotally, refcounting is written wrong even more commonly than binary search is 😆